File: java-virtual-threads-demo/spring-webflux-rest-call/src/main/java/com/java/programming/springwebfluxrestcall/controller/EmployeeRestController.java
package com.java.programming.springwebfluxrestcall.controller;

import com.java.programming.springwebfluxrestcall.model.Employee;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;
import org.springframework.web.reactive.function.client.WebClient;
import reactor.core.publisher.Flux;

@RestController
@RequiredArgsConstructor
@RequestMapping
public class EmployeeRestController {

    private final WebClient employeeRestClient;

    @GetMapping("/get-all-employees")
    public Flux<Employee> callSlowServer(){
        return employeeRestClient.get()
                .uri("/employees")
                .retrieve()
                .bodyToFlux(Employee.class);
    }
}

File: java-virtual-threads-demo/kotlin-coroutines-rest-call/src/main/kotlin/com/java/programming/kotlincoroutinesrestcall/controller/EmployeeRestController.kt
package com.java.programming.kotlincoroutinesrestcall.controller

import com.java.programming.kotlincoroutinesrestcall.model.Employee
import kotlinx.coroutines.flow.Flow
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController
import org.springframework.web.reactive.function.client.*

@RestController
@RequestMapping
class EmployeeRestController {

    @Autowired
    lateinit var employeeWebClient: WebClient

    @GetMapping("/get-all-employees")
    suspend fun callSlowServer(): Flow<Employee> {
        return employeeWebClient.get()
                .uri("/employees")
                .exchangeToFlow { clientResponse ->  clientResponse.bodyToFlow<Employee>() }
        }
}